### PR TITLE
[Browser] Goodbye OWS. Fixes #24841

### LINKED ISSUE #24841: Please reomve the not functional OWS group from the QGIS browser
Author Name: **Giovanni Manghi** (@gioman)
Original Redmine Issue: [16942](https://issues.qgis.org/issues/16942)
Affected QGIS version: 3.3(master)
Redmine category:browser

---
Possibly also affects WFS and WCS.

The connection names show, they do not expand to show layers.

Tested with Geoserver, Mapserver and QGIS Server.

I believe is a regression.

### PR DESCRIPTION
**Pull request by lbartoletti**
## Description

Every time I want to add a connection, I get confused and want to use this OWS menu. I know that it is useless, others says so too ( #24841 ).

It is no longer present in the browser, but it is still functional and no longer adds confusion.

image

### CODE REVIEW

**Comment by gioman** [comment]
@lbartoletti thanks!

File: src/core/browser/qgsdataitemproviderregistry.cpp [hunk 1]
```diff
@@ -25,6 +25,12 @@ QgsDataItemProviderRegistry::QgsDataItemProviderRegistry()
 {
   QStringList providersList = QgsProviderRegistry::instance()->providerList();
 
+  // OWS is not useful in the browser.
+  // The OWS type connections (WFS/WMS) are available in their respective menus.
+  // Having an OWS menu in the browser only brings disadvantages.
+  // See https://github.com/qgis/QGIS/issues/24841
+  providersList.removeOne( QStringLiteral( "ows" ) );
```
Comment (nyalldawson) [inline src/core/browser/qgsdataitemproviderregistry.cpp:32]: This is an odd way to remove it -- it's still being created and then deleted. Better would be to just remove QgsOwsProviderMetadata::dataItemProviders() and qgsowsdataitem.h/.cpp entirely.

Comment (nyalldawson) [reply src/core/browser/qgsdataitemproviderregistry.cpp:32]: Actually just remove the whole src/providers/ows folder, none of that is required

Comment (lbartoletti) [reply src/core/browser/qgsdataitemproviderregistry.cpp]: Indeed.

note: I spent a few minutes wondering why ows was still appearing in the browser even though I had deleted its code... libowsprovider.so was still in my build folder and therefore remained present. :sweat_smile:

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**